runtime/docker/docker.go

```diff
@@ -714,7 +714,7 @@ func (d *DockerRuntime) DeleteContainer(ctx context.Context, cID string) error {
 		}
 	}
 	log.Debugf("Removing container: %s", strings.TrimLeft(cID, "/"))
-	err = d.Client.ContainerRemove(ctx, cID, dockerTypes.ContainerRemoveOptions{Force: force})
+	err = d.Client.ContainerRemove(ctx, cID, dockerTypes.ContainerRemoveOptions{Force: force, RemoveVolumes: true})
```

Comment:
@rudranil-das @ashutshkumr 
FYI, I have added removal of volume mounts and thus removed the limitation#2 from the docs.

Reply:
That's awesome, thanks !